Make it possible to use the bot to send notifications

mbaynton opened the pull request:
This makes it possible to use the bot to send notifications about things happening outside the visibility of the bot's own commands. `go-chat-bot` provides nice abstractions for sending messages to IRC or Slack, it's just hard to get at the raw send function because it's not exported.

So this adds Bot.SendUnsolicitedMessage that can easily be called by any other code in a larger program to send some notification. It also makes it possible to get a hold of the bot instance by adding an `irc.SetUp()` func returning the bot that can optionally be called before irc.Run().

Only modified the IRC package for now, will modify the others if the concept / PR receives positive feedback.

fabioxgn commented:
@mbaynton I don't think that "Unsolicited" is a good name for this, maybe just "Send message". Also I think that the user value should be ommited, as the user will be always the bot, right?

mbaynton commented:
Thanks for the look, @fabioxgn. I've renamed the function.

It occurred to me that if there's going to be a function called bot.SendMessage, it's probably how all messages should be sent, no matter whether they happen to come from code inside the bot package or not. So, I updated existing calls to b.handlers.Response() to use SendMessage(). This means the sender needs to stay.

Another possible way keeping sender could be useful to someone: I think it'd be easy now to create for example an irc bot and a slack bot running in separate goroutines, and create a relay.
